### ISSUE TITLE
certbot-auto 0.16.0 crashes when resizing putty window at prompt

### ISSUE THREAD

**Issue by Philippe23** [certbot/certbot#4926]
certbot-auto 0.16.0 crashes when resizing putty window at prompt.  Not sure if putty is a requirement, but haven't had a chance to try from another machine.

Also not sure if https://github.com/certbot/certbot/issues/3030 is related or not.  I don't have dialog installed so I can't test to see if it crashes as mentioned in issue #3030.

## My operating system is (include version):

Server: Amazon Linux (amzn-ami-hvm-2016.09.0.20160923-x86_64-gp2)
Client: Windows 10

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):

certbot-auto

## I ran this command and it produced this output:

sudo /root/certbot-auto certonly --debug --manual --preferred-challenges dns-01  -d yyyy.xxxxx.com

## Certbot's behavior differed from what I expected because:

Don't expect a window resize to cause a crash/exit.

### Console output:
```
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Cert is due for renewal, auto-renewing...
Renewing an existing certificate
Performing the following challenges:
dns-01 challenge for yyyy.xxxxx.com

-------------------------------------------------------------------------------
NOTE: The IP of this machine will be publicly logged as having requested this
certificate. If you're running certbot in manual mode on a machine that is not
your server, please ensure you're okay with that.

Are you OK with your IP being logged?
-------------------------------------------------------------------------------
(Y)es/(N)o: Cleaning up challenges
Exiting abnormally:
Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/bin/letsencrypt", line 11, in <module>
    sys.exit(main())
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 743, in main
    return config.func(config, plugins)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 683, in certonly
    lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 77, in _get_and_save_cert
    renewal.renew_cert(config, domains, le_client, lineage)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/renewal.py", line 297, in renew_cert
    new_certr, new_chain, new_key, _ = le_client.obtain_certificate(domains)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/client.py", line 317, in obtain_certificate
    self.config.allow_subset_of_names)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/auth_handler.py", line 74, in get_authorizations
    resp = self._solve_challenges()
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/auth_handler.py", line 115, in _solve_challenges
    resp = self.auth.perform(self.achalls)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/plugins/manual.py", line 140, in perform
    self._verify_ip_logging_ok()
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/plugins/manual.py", line 167, in _verify_ip_logging_ok
    if display.yesno(msg, cli_flag=cli_flag, force_interactive=True):
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/display/util.py", line 220, in yesno
    no=_parens_around_char(no_label)))
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/display/util.py", line 79, in input_with_timeout
    rlist, _, _ = select.select([sys.stdin], [], [], timeout)
error: (4, 'Interrupted system call')
Please see the logfiles in /var/log/letsencrypt for more details.
```

### Logfile ###
```
2017-07-13 16:22:45,549:DEBUG:certbot.main:certbot version: 0.16.0
2017-07-13 16:22:45,549:DEBUG:certbot.main:Arguments: ['--debug', '--manual', '--preferred-challenges', 'dns-01', '-d', 'yyyy.xxxxx.com']
2017-07-13 16:22:45,549:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2017-07-13 16:22:45,567:DEBUG:certbot.log:Root logging level set at 20
2017-07-13 16:22:45,567:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2017-07-13 16:22:45,568:DEBUG:certbot.plugins.selection:Requested authenticator manual and installer None
2017-07-13 16:22:45,571:DEBUG:certbot.plugins.selection:Single candidate plugin: * manual
Description: Manual configuration or run your own shell scripts
Interfaces: IAuthenticator, IPlugin
Entry point: manual = certbot.plugins.manual:Authenticator
Initialized: <certbot.plugins.manual.Authenticator object at 0x7f795083b790>
Prep: True
2017-07-13 16:22:45,572:DEBUG:certbot.plugins.selection:Selected authenticator <certbot.plugins.manual.Authenticator object at 0x7f795083b790> and installer None
2017-07-13 16:22:45,576:DEBUG:certbot.main:Picked account: <Account(RegistrationResource(body=Registration(status=None, contact=(u'mailto:pchaintreuil@sanzaru.com',), agreement=u'https://letsencrypt.org/documents/LE-SA-v1.1.1-August-1-2016.pdf', key=JWKRSA(key=<ComparableRSAKey(<cryptography.hazmat.backends.openssl.rsa._RSAPublicKey object at 0x7f79503a7bd0>)>)), uri=u'https://acme-v01.api.letsencrypt.org/acme/reg/11185620', new_authzr_uri=u'https://acme-v01.api.letsencrypt.org/acme/new-authz', terms_of_service=u'https://letsencrypt.org/documents/LE-SA-v1.1.1-August-1-2016.pdf'), 1818eb936c51e85fd33793ccea137b09, Meta(creation_host=u'ip-172-31-0-242.us-west-2.compute.internal', creation_dt=datetime.datetime(2017, 3, 21, 18, 15, 49, tzinfo=<UTC>)))>
2017-07-13 16:22:45,577:DEBUG:acme.client:Sending GET request to https://acme-v01.api.letsencrypt.org/directory.
2017-07-13 16:22:45,580:DEBUG:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
2017-07-13 16:22:45,814:DEBUG:requests.packages.urllib3.connectionpool:https://acme-v01.api.letsencrypt.org:443 "GET /directory HTTP/1.1" 200 352
2017-07-13 16:22:45,815:DEBUG:acme.client:Received response:
HTTP 200
Server: nginx
Content-Type: application/json
Content-Length: 352
Boulder-Request-Id: IBf3cZ696QoVRN38Uul4ExDSMwp5pcJAy17IWKG3hK4
Replay-Nonce: bV0cQsPTiiINtI9aZyGHwU4SCj7UiqMmb6UL7K0BgAY
X-Frame-Options: DENY
Strict-Transport-Security: max-age=604800
Expires: Thu, 13 Jul 2017 16:22:45 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Thu, 13 Jul 2017 16:22:45 GMT
Connection: keep-alive

{
  "key-change": "https://acme-v01.api.letsencrypt.org/acme/key-change",
  "new-authz": "https://acme-v01.api.letsencrypt.org/acme/new-authz",
  "new-cert": "https://acme-v01.api.letsencrypt.org/acme/new-cert",
  "new-reg": "https://acme-v01.api.letsencrypt.org/acme/new-reg",
  "revoke-cert": "https://acme-v01.api.letsencrypt.org/acme/revoke-cert"
}
 2017-07-13 16:22:45,821:DEBUG:certbot.storage:Should renew, less than 30 days before certificate expiry 2017-08-08 13:34:00 UTC.
2017-07-13 16:22:45,822:INFO:certbot.renewal:Cert is due for renewal, auto-renewing...
2017-07-13 16:22:45,822:INFO:certbot.main:Renewing an existing certificate
2017-07-13 16:22:45,822:DEBUG:acme.client:Requesting fresh nonce
2017-07-13 16:22:45,822:DEBUG:acme.client:Sending HEAD request to https://acme-v01.api.letsencrypt.org/acme/new-authz.
2017-07-13 16:22:45,916:DEBUG:requests.packages.urllib3.connectionpool:https://acme-v01.api.letsencrypt.org:443 "HEAD /acme/new-authz HTTP/1.1" 405 0
2017-07-13 16:22:45,917:DEBUG:acme.client:Received response:
HTTP 405
Server: nginx
Content-Type: application/problem+json
Content-Length: 91
Allow: POST
Boulder-Request-Id: duXFAHa0UCxN2brjHmSJEqgL99OpgxPFeL96sTBsJbk
Replay-Nonce: g5A2lsqvUc8eUIPjTLilst9qwFm7ib4xxdIWjZllWdo
Expires: Thu, 13 Jul 2017 16:22:45 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Thu, 13 Jul 2017 16:22:45 GMT
Connection: keep-alive


2017-07-13 16:22:45,917:DEBUG:acme.client:Storing nonce: g5A2lsqvUc8eUIPjTLilst9qwFm7ib4xxdIWjZllWdo
2017-07-13 16:22:45,918:DEBUG:acme.client:JWS payload:
{
  "identifier": {
    "type": "dns",
    "value": "yyyy.xxxxx.com"
  },
  "resource": "new-authz"
}
2017-07-13 16:22:45,921:DEBUG:acme.client:Sending POST request to https://acme-v01.api.letsencrypt.org/acme/new-authz:
{
  "protected": "eyJub25jZSI6ICJnNUEybHNxdlVjOGVVSVBqVExpbHN0OXF3Rm03aWI0eHhkSVdqWmxsV2RvIiwgImFsZyI6ICJSUzI1NiIsICJqd2siOiB7ImUiOiAiQVFBQiIsICJrdHkiOiAiUlNBIiwgIm4iOiAieG1nNmdsQzd1Y25MRm8xc0txNU5QZWtieE4xQXNReHlOUE9pQU05VzlYcjhJNE4ybGxsd3lvWnVYZm5IaVQ5cHFaellTYXFlWTd4SWtaajNFbXg5eU4xSmo5RXNVbzlib0ZpeEh2cEFQMHd4VG5mSXVxTUV1YkFWanZRRm50VmxhUVhrZUU2SFZMWGNxMjVPbnJIcWtFNXF3dzI2cmFZY2xtM2lsQnBLV1pJZ1p6QnVOUVVUTlR4eDBVY2xJV2xuRXhVYjZuM3lILTJBRmFhQTVJT1NLRE8xUTVxbFJfdm1HSER5YlBNX2JFc3lVNDQyLWJHeUw1a1FTR09NaF93NDZfZHNBMVVVWXFDaHBLVGM4TFNKcllNaExJUkc3MzBPR3ZieU9OemItLTVjdXV4ODJWMEgtUHhoVEFTajl3d1MzZHZtcnVVVEFmVXh0bkk1dWtDa1F3In19",
  "payload": "ewogICJpZGVudGlmaWVyIjogewogICAgInR5cGUiOiAiZG5zIiwgCiAgICAidmFsdWUiOiAiYnVpbGQtMjAxNzAzMjEtMDAxLmFwaS5yaXBjb2lsLmNvbSIKICB9LCAKICAicmVzb3VyY2UiOiAibmV3LWF1dGh6Igp9",
  "signature": "gjdKW4Lr8Lm8lKK0oNG5t5fUXWt-wPkQ5BbehwW-vdTnlgDM0i2sFmGLbydPkl8u01YeEXt9Rqo45dZ1kkmDYYoRXkBizZRc7mOjAFYhciwHth4-h-LtR2jkVIgjZJeU8o8ypHyDOxHl7SUBr8Rrynz1C3eoBs3wV--D5LP5k5iwdJleozhpOFuxHqUqM6XeQmF-_yV2ktGBJES_7yKYOHnkZpzBCgLdOjKrRXvmwmdjEQHHOrgUwK0oLGM_267fxWsQYaQgwwskwGLgP8ILfm1Qn8d5wH7Bzji-9sekA7Z0-k31px2D6hHf7t6UelLghIm3MfGcR4Deh2LOD0_CGQ"
}
2017-07-13 16:22:46,065:DEBUG:requests.packages.urllib3.connectionpool:https://acme-v01.api.letsencrypt.org:443 "POST /acme/new-authz HTTP/1.1" 201 1022
2017-07-13 16:22:46,065:DEBUG:acme.client:Received response:
HTTP 201
Server: nginx
Content-Type: application/json
Content-Length: 1022
Boulder-Request-Id: aqeEyrz3SpfAUtKTr1Tb8rswetUcJe7mBFRoOq7V58c
Boulder-Requester: 11185620
Link: <https://acme-v01.api.letsencrypt.org/acme/new-cert>;rel="next"
Location: https://acme-v01.api.letsencrypt.org/acme/authz/hEuU6vbrd4XUCSm8s_ZgGvVnBRtXnYplSeKwKxWECQE
Replay-Nonce: 2lMoaDRV6SoRsS9lZlFPHUPcLj06ZZGHFus0QtW0PSI
X-Frame-Options: DENY
Strict-Transport-Security: max-age=604800
Expires: Thu, 13 Jul 2017 16:22:46 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Thu, 13 Jul 2017 16:22:46 GMT
Connection: keep-alive

{
  "identifier": {
    "type": "dns",
    "value": "yyyy.xxxxx.com"
  },
  "status": "pending",
  "expires": "2017-07-20T16:22:46.025976642Z",
  "challenges": [
    {
      "type": "dns-01",
      "status": "pending",
      "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/hEuU6vbrd4XUCSm8s_ZgGvVnBRtXnYplSeKwKxWECQE/1538024536",
      "token": "QlTHUtt_lZ7dMuhXLTmp5mZGL5_DDkwg-2PVqktvITQ"
    },
    {
      "type": "tls-sni-01",
      "status": "pending",
      "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/hEuU6vbrd4XUCSm8s_ZgGvVnBRtXnYplSeKwKxWECQE/1538024537",
      "token": "kSxHVlPKxmSyi0P7rjmbGqKMT-2JS_lK-OZQtvKuTRo"
    },
    {
      "type": "http-01",
      "status": "pending",
      "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/hEuU6vbrd4XUCSm8s_ZgGvVnBRtXnYplSeKwKxWECQE/1538024538",
      "token": "ID-yyD_HaNL0QtTL_01hTJAef3MpCZ0Eq-0IhuwLinY"
    }
  ],
  "combinations": [
    [
      1
    ],
    [
      2
    ],
    [
      0
    ]
  ]
}
2017-07-13 16:22:46,066:DEBUG:acme.client:Storing nonce: 2lMoaDRV6SoRsS9lZlFPHUPcLj06ZZGHFus0QtW0PSI
2017-07-13 16:22:46,066:INFO:certbot.auth_handler:Performing the following challenges:
2017-07-13 16:22:46,066:INFO:certbot.auth_handler:dns-01 challenge for yyyy.xxxxx.com
2017-07-13 16:22:46,917:DEBUG:certbot.error_handler:Encountered exception:
Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/auth_handler.py", line 115, in _solve_challenges
    resp = self.auth.perform(self.achalls)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/plugins/manual.py", line 140, in perform
    self._verify_ip_logging_ok()
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/plugins/manual.py", line 167, in _verify_ip_logging_ok
    if display.yesno(msg, cli_flag=cli_flag, force_interactive=True):
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/display/util.py", line 220, in yesno
    no=_parens_around_char(no_label)))
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/display/util.py", line 79, in input_with_timeout
    rlist, _, _ = select.select([sys.stdin], [], [], timeout)
error: (4, 'Interrupted system call')

2017-07-13 16:22:46,917:DEBUG:certbot.error_handler:Calling registered functions
2017-07-13 16:22:46,917:INFO:certbot.auth_handler:Cleaning up challenges
2017-07-13 16:22:46,917:ERROR:certbot.log:Exiting abnormally:
Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/bin/letsencrypt", line 11, in <module>
    sys.exit(main())
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 743, in main
    return config.func(config, plugins)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 683, in certonly
    lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 77, in _get_and_save_cert
    renewal.renew_cert(config, domains, le_client, lineage)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/renewal.py", line 297, in renew_cert
    new_certr, new_chain, new_key, _ = le_client.obtain_certificate(domains)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/client.py", line 317, in obtain_certificate
    self.config.allow_subset_of_names)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/auth_handler.py", line 74, in get_authorizations
    resp = self._solve_challenges()
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/auth_handler.py", line 115, in _solve_challenges
    resp = self.auth.perform(self.achalls)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/plugins/manual.py", line 140, in perform
    self._verify_ip_logging_ok()
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/plugins/manual.py", line 167, in _verify_ip_logging_ok
    if display.yesno(msg, cli_flag=cli_flag, force_interactive=True):
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/display/util.py", line 220, in yesno
    no=_parens_around_char(no_label)))
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/display/util.py", line 79, in input_with_timeout
    rlist, _, _ = select.select([sys.stdin], [], [], timeout)
error: (4, 'Interrupted system call')
```

**Comment by SwartzCr** [comment]
#4871 seems to be related
It seems to be basically this issue? https://stackoverflow.com/questions/19348236/ncurses-interrupts-system-call-when-resizing-terminal
Which suggests that we can just catch and ignore this error. Since I've never messed with ncurses code I'd like some feedback from @erikrose who has to see if this is a good idea or not. It doesn't seem worthwhile to send this upstream, so hopefully we can make a bugfix for this.

**Comment by bmw** [comment]
While I really appreciate the detailed post here, this is a duplicate of #4871 so I'm closing this issue.

**Comment by bmw** [comment]
This is due to Python's `select.select` used [here](https://github.com/certbot/certbot/blob/d4676610e9009397000522cc1fe5420242905e86/certbot/display/util.py#L79) raising an exception when the terminal is resized.

**Comment by SwartzCr** [comment]
Possibly fixed in #5001

**Comment by sydneyli** [comment]
@bmw, do you happen to know if this issue was fixed in #5001?

**Comment by bmw** [comment]
I don't think this issue has been fixed. It looks like that PR was closed rather than merged and I just reproduced this issue by waiting for Certbot to prompt me and then resized my terminal.

**Comment by bmw** [comment]
This issue has been closed due to lack of activity, but if you think it should be reopened, please open a new issue with a link to this one and we'll take a look.

**Comment by bmw** [comment]
We got another report of this at https://github.com/certbot/certbot/issues/7637.

alexzorin says that this should be fixed when we drop support for Python 2.7 and 3.4 though.
